connection.go

```diff
@@ -83,6 +83,9 @@ func (c *connection) sendRequest(req proto.Request, reqID int32) (*bytes.Reader,
 	}()
 	select {
 	case result := <-readRespChan:
+		if result.err != nil {
+			c.Close()
```

Comment:
I believe this is correct but also it's the kind of thing that might have side effects/changes in behavior. Just wanted to call it out.

Reply:
Does it always indicate an issue with the connection?  I think it could also be a malformed response, which isn't connection.go's problem?

Reply:
I agree it could be a malformed response, but in that case what do we do with the connection? Can we trust the rest of the stream to be aligned along message boundaries?

I'm thinking about the case where we get a flipped bit or something and we start processing garbage. We send a request, we get a garbage response, what would we do? Send another request? It might work or it might now.

I think the cost of establishing a new connection is relatively small, so in this case I thought it worth erring on the side of closing in the face of any uncertainty.

Thoughts?

Reply:
basically I don't know how often a malformed response warrants re-creating the connection, it can't happen that often?  I'll leave this up to you